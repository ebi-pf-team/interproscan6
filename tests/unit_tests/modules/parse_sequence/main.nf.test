nextflow_process {

    name "Test Process PARSE_SEQUENCE"
    script "interproscan/modules/parse_sequence/main.nf"
    process "PARSE_SEQUENCE"

    stage {
        symlink "interproscan/scripts/parse_sequence/parse_sequence.py"
        symlink "tests/unit_tests/test_inputs/parse_sequences/allowed.faa"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/parse_sequences/allowed.faa")
                input[1] = file("meta/tests/unit_tests/test_inputs/parse_sequences/allowed.faa")
                input[2] = "false"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
