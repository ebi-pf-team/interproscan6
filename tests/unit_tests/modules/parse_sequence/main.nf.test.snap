{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "parsed_sequences:md5,52f337742b16b255d755a41a4d640874"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-17T14:09:34.557144"
    }
}